scss/util/_value.scss

```diff
@@ -138,3 +138,23 @@
   @return if(type-of($map) != 'list', ($value,), $map);
 
 }
+
+/// Safely return a value from a map.
+///
+/// @param {Map} $map - Map to retrieve a value from.
+/// @param {String} $name - Name of the map key.
+///
+/// @returns {List} Found value.
+@function map-safe-get($map, $name) {
+  @if (type-of($map) == 'map') {
```

Comment:
Sorry not sure I understand.
You want to replace `@if (type-of($map) == 'map') {`
with `@if (type-of($map) == 'map' or (type-of($map) == 'list' and length($map) == 0)) {`
?